app/controllers/road-sign-concepts/road-sign-concept.js

```diff
@@ -1,32 +1,71 @@
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
-import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
 
 export default class RoadsignConceptsRoadsignConceptController extends Controller {
-  @service router;
-  @service intl;
+  @tracked isToShowSubSigns = false;
+  @tracked isToShowRelated = false;
+
+  @tracked category = null;
 
   @action
-  async removeRoadSignConcept(roadSignConcept, event) {
+  async editSubSignRelation(subSign, event) {
     event.preventDefault();
-
-    await roadSignConcept.destroyRecord();
-    this.router.transitionTo('road-sign-concepts');
+    let subSigns = await this.model.roadSignConcept.subSigns;
+    let roadSigns = await this.model.allSubSigns.roadSignConcepts;
+    if (event.target.checked) {
+      subSigns.pushObject(subSign);
+      this.model.roadSignConcept.save();
+      roadSigns.removeObject(subSign);
+      this.model.allSubSigns.save();
+    } else {
+      subSigns.removeObject(subSign);
+      this.model.roadSignConcept.save();
+      roadSigns.pushObject(subSign);
+      this.model.allSubSigns.save();
+    }
   }
 
   @action
-  async removeSubSign(subSign, event) {
+  async editRelatedRoadSignRelation(relatedRoadSign, event) {
     event.preventDefault();
-    let subSigns = await this.model.subSigns;
-    subSigns.removeObject(subSign);
-    this.model.save();
+    let relatedRoadSigns = await this.model.roadSignConcept
+      .relatedRoadSignConcepts;
+    if (event.target.checked) {
+      relatedRoadSigns.pushObject(relatedRoadSign);
+      this.model.roadSignConcept.save();
+    } else {
+      relatedRoadSigns.removeObject(relatedRoadSign);
+      this.model.roadSignConcept.save();
+    }
   }
 
   @action
-  async removeRelatedRoadSignConcept(relatedRoadSignConcept, event) {
-    event.preventDefault();
-    let relatedRoadSignConcepts = await this.model.relatedRoadSignConcepts;
-    relatedRoadSignConcepts.removeObject(relatedRoadSignConcept);
-    this.model.save();
+  showEditSubSigns() {
+    this.isToShowSubSigns = !this.isToShowSubSigns;
+    if (this.isToShowRelated) this.isToShowRelated = false;
+  }
+
+  @action
+  showEditRelatedRoadSigns() {
+    this.isToShowRelated = !this.isToShowRelated;
+    if (this.isToShowSubSigns) this.isToShowSubSigns = false;
+  }
+
+  get showSidebar() {
+    return this.isToShowRelated || this.isToShowSubSigns;
+  }
+
+  get filteredCategoryLabel() {
```

Comment:
I'm not sure this one is needed? It seems that it finds the currently selected category inside the list of all categories and returns it? Why not use the selected category directly? I think it will have the same effect?

Reply:
No, it returns the road signs of the selected category.

Reply:
Ah yeah, it returns the category that I'm searching but what I take is the road signs list.